File: 1514/Solution.java
class Solution {
    public double maxProbability(int n, int[][] edges, double[] succProb, int start_node, int end_node) {
        double[] max = new double[n];
        max[start_node] = 1.0;
        for(int i = 0; i < n - 1; i++){
            boolean updated = false;
            for(int j = 0; j < edges.length; j++){
                int u = edges[j][0];
                int v = edges[j][1];
                double prob = succProb[j];

                if(max[u] * prob > max[v]){
                    max[v] = max[u] * prob;
                    updated = true;
                }
                if(max[v] * prob > max[u]){
                    max[u] = max[v] * prob;
                    updated = true;
                }
            }    
            if(!updated) break;
        }    
        return max[end_node];
    }
}
// time O(n * E)
// space O(n)

File: 1514/Solution.go
func maxProbability(n int, edges [][]int, succProb []float64, start_node int, end_node int) float64 {
    max := make([]float64, n)
    max[start_node] = 1.0
    for i:=0; i< n-1; i++{
        updated := false
        for j:=0; j< len(edges); j++{
            u, v, prob := edges[j][0], edges[j][1], succProb[j]
            if max[u] * prob > max[v] {
                max[v] = max[u] * prob
                updated = true
            }
            if max[v] * prob > max[u] {
                max[u] = max[v] * prob
                updated = true
            }
        }
        if !updated {
            break
        }
    }
    return max[end_node]
}
